boopmain.ino

```diff
@@ -128,10 +142,20 @@ void loop()
     draw();
     vibe_flash();
     last_draw = millis();
-        
-    if(color_select > 6) {
+  }
+
+//Color change if button is released
+  if(is_button_pressed == false & last_buttonpress == true) {
+    color_select++;
+  }
+  //Update last button press
+  if(is_button_pressed){
+    last_buttonpress = true;
+} else {
+    last_buttonpress = false;
+}
+//Cycle back through the list of colours
+  if(color_select > 19) {
```

Comment:
be careful here, high chance of this being a future bug.

Setting a variable called 'COLOR_COUNT' might make it easier to keep the value updated